### DISCUSSION TITLE
When to use Tables vs Data Grid

### DISCUSSION THREAD

**Discussion by brannytempy** [twilio-labs/paste#2729] [Ideas] [answered]
Hi Paste!

As Segment prepares to onboard to Paste, one of the biggest gaps in documentation I see is when to use a Table vs a Data Grid. Both pages have a `When to use ...`, which is great! Where I struggle is the `interactive` guidance, and what specifically is considered an interaction.

What I'd like to see is more concrete guidance in this area. For example, [I found a discussion that mentioned your recommendation is to not have links or buttons in a Table](https://github.com/twilio-labs/paste/discussions/887). It would be helpful to flow as much guidance to these pages to ensure the right component is used.

In terms of guidance, it would be great to list out which interactions should guide you to which component. Such as:
- Filtering
- Searching
- Sorting
- Actions inside of cells
- Pagination
- Show / Hide Columns
- Any other considerations

Thank you!! ʕ•ᴥ•ʔ

**Comment by nkrantz** [comment]
Hey @brannytempy - we're bringing your 2 discussions up as a team and we'll update the threads soon. Thanks for sharing your feedback, more to come!

**Comment by TheSisb** [comment, marked as answer]
Hi @brannytempy,

Good call, I'll make a ticket in our backlog to add something to those doc pages.

While I'm here, one idea to help you frame your thought process around which to use is to consider keyboard navigation.  In a Table, every focusable element is a tab stop. You can imagine that a Table with several rows of buttons or links would be tedious to traverse.  On the other hand, DataGrid prevents tabbing through child nodes until [you activate `Actionable` mode](https://paste.twilio.design/components/data-grid#keyboard-navigation). This makes DataGrid more suitable for data with interaction.

Hope this helps a bit.

**Reply by gloriliale** [reply]
Update: the [ticket for this is here](https://issues.corp.twilio.com/browse/DSYS-3632) if you'd like to follow along!
